core/primitives/version.hpp

```diff
@@ -118,11 +132,21 @@ namespace kagome::primitives {
   Stream &operator>>(Stream &s, Version &v) {
     s >> v.spec_name >> v.impl_name >> v.authoring_version >> v.spec_version
         >> v.impl_version >> v.apis;
-    // old Kusama runtimes do not contain transaction_version
-    if (s.hasMore(sizeof(v.transaction_version))) {
+
+    auto core_version = detail::coreVersionFromApis(v.apis);
+    // old Kusama runtimes do not contain transaction_version and state_version
+    // https://github.com/paritytech/substrate/blob/1b3ddae9dec6e7653b5d6ef0179df1af831f46f0/primitives/version/src/lib.rs#L238
+    if (core_version.has_value() and core_version.value() >= 3
+        and s.hasMore(sizeof(v.transaction_version))) {
       s >> v.transaction_version;
     } else {
-      v.transaction_version = 0;
+      v.transaction_version = 1;
```

Comment:
Yes, seems this [had to be](https://github.com/paritytech/substrate/blob/1b3ddae9dec6e7653b5d6ef0179df1af831f46f0/primitives/version/src/lib.rs#L238) 1.